src/scripts/party/Party.ts

```diff
@@ -40,21 +40,47 @@ class Party implements Feature {
 
     }
 
-    gainPokemonByName(name: PokemonNameType, shiny = false, suppressNotification = false, gender = -1) {
+    gainPokemonByName(name: PokemonNameType, shiny = false, suppressNotification = false, gender = -1, shadow = GameConstants.ShadowStatus.None) {
         const pokemon = pokemonMap[name];
-        this.gainPokemonById(pokemon.id, shiny, suppressNotification, gender);
+        this.gainPokemonById(pokemon.id, shiny, suppressNotification, gender, shadow);
     }
 
-    gainPokemonById(id: number, shiny = false, suppressNotification = false, gender = -1) {
+    gainPokemonById(id: number, shiny = false, suppressNotification = false, gender = -1, shadow = GameConstants.ShadowStatus.None) {
         // If no gender defined, calculate it
         if (gender === -1) {
             gender = PokemonFactory.generateGenderById(id);
         }
-        this.gainPokemon(PokemonFactory.generatePartyPokemon(id, shiny, gender), suppressNotification);
+        this.gainPokemon(PokemonFactory.generatePartyPokemon(id, shiny, gender, shadow), suppressNotification);
     }
 
     gainPokemon(pokemon: PartyPokemon, suppressNotification = false) {
         PokemonHelper.incrementPokemonStatistics(pokemon.id, GameConstants.PokemonStatisticsType.Captured, pokemon.shiny, pokemon.gender);
+
+        if (pokemon.shadow) {
+            // Already caught (shadow)
+            if (this.alreadyCaughtPokemon(pokemon.id, false, true)) {
+                return;
```

Comment:
If you don't have a pokemon shadow but not shiny, and then you catch a shiny shadow, this looks like it will prevent you getting shiny credit.

Reply:
Fixed